plugins/fontconfig.lua

```diff
@@ -0,0 +1,57 @@
+-- mod-version:1 -- lite-xl 2.00
+local subprocess = require "process"
+
+local core = require "core"
+local style = require "core.style"
+local config = require "core.config"
+
+--[[
+    Example config (put it in user module):
+    ```
+    local fc = require "plugins.fontconfig"
+    fc(
+        { name = "sans", size = 13 * SCALE },     -- UI font
```

Comment:
Why only supports UI and code fonts ?

This looks too much ad hoc. I was wondering if we cannot do something like:

```lua
style.font = fontconfig.load("sans", 13 * SCALE)
-- do the same of above for any font the application may use
```

Reply:
That's what I'd like it to do, but I don't want the process to block lite completely. At this point the editor hasn't even started yet, so if `fc-match` chokes up, the users will be staring into a grey screen.

Reply:
We can also expose a asynchronous version, say:
```lua
core.add_thread(function()
    style.font = fontconfig.load_async("sans", 13 * SCALE)
end)
```

That's what the function does anyway.